crates/ruff/src/rules/flake8_bandit/helpers.rs

```diff
@@ -1,10 +1,14 @@
+use once_cell::sync::Lazy;
+use regex::Regex;
 use rustpython_parser::ast::{Constant, Expr, ExprKind};
 
 use crate::checkers::ast::Checker;
 
-const PASSWORD_NAMES: [&str; 7] = [
-    "password", "pass", "passwd", "pwd", "secret", "token", "secrete",
-];
+static PASSWORD_CANDIDATE_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(
+    r"(^(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)$|_(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)_|^(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)_|_(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)$)"
```

Comment:
There is absolutely no need to repeat `(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)` four times within this regex.

Reply:
I think the regex could be simplified to `\b(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)\b`.

Reply:
I don't fully understand why they wrote it out in that way, but what @edgarrmondragon included here looks like a direct port of the version in Bandit: https://github.com/PyCQA/bandit/blob/a9eaafa53ae73c1a66172373128cb19bb7b82a32/bandit/plugins/general_hardcoded_password.py#L13-L16

Reply:
You're right @charliermarsh, I just ported the Python regex.

@not-my-profile thanks for the suggestion, it makes total